papas/detectors/CMS.py

```diff
@@ -32,34 +38,57 @@ def acceptance(self, cluster):
         eta = abs(cluster.position.Eta())
         if eta < self.eta_crack:
             return energy>self.emin
-        elif eta < 3.:
-            return energy>self.emin and cluster.pt>0.5
+        elif eta < 2.93:
+            return energy>1 and cluster.pt>0.2
         else:
             return False
 
     def space_resolution(self, ptc):
         pass
-
     
 class HCAL(DetectorElement):
 
     def __init__(self):
         volume = VolumeCylinder('hcal', 2.9, 3.6, 1.9, 2.6 )
         mat = material.Material('HCAL', None, 0.17)
-        self.eres = [1.1, 0., 0.]
+        self.eres = {'barrel':[1.25829, 0., 0.175950], 'endcap':[1.32242e-06, 6.99123, 2.70281e-01]}
+        self.eresp = {'barrel':[1.03430, 5.23646, -2.03400], 'endcap':[1.06742, 9.41242, -2.75191]}
         super(HCAL, self).__init__('ecal', volume, mat)
 
-    def energy_resolution(self, energy, theta=0.):
-        return self.eres[0]/ math.sqrt( energy ) 
+    def energy_resolution(self, energy, eta=0.):
+        part = 'barrel'
+        if abs(eta)>1.3 and abs(eta)<3.0:
+            part = 'endcap'
+        stoch = self.eres[part][0] / math.sqrt(energy)
+        noise = self.eres[part][1] / energy
+        constant = self.eres[part][2]
+        return math.sqrt( stoch**2 + noise**2 + constant**2)
+
+    def energy_response(self, energy, eta=0):
+        part = 'barrel'
+        if abs(eta)>1.3 and abs(eta)<3.0:
```

Comment:
I don't like the duplication of hardcoded parameters. if parameter used more than once, e.g. 1.3, set this param as attribute in constructor